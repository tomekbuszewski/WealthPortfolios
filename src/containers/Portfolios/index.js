import * as React from 'react';
import Communication from '@src/services/Communication';

import Loader from '@src/components/Loader';
import View from '@src/views/Portfolios';

import {
  ENDPOINTS,
  ROUTES,
} from '@config';

class Portfolios extends React.Component {
  static createModel(data) {
    return data.map(item => ({
      ...item,
      key: `${item.clientName}-${item.name}`,
      to: `${ROUTES.PORTFOLIOS}${item.id}`,
    }));
  }

  constructor(props) {
    super(props);

    this.fetch = new Communication({
      url: ENDPOINTS.PORTFOLIOS,
    });

    this.state = {
      portfolios: [],
    };
  }

  componentDidMount() {
    this.fetch().then(({ data }) => this.setState({
      portfolios: Portfolios.createModel(data),
    }));
  }

  render() {
    const { portfolios } = this.state;
    return portfolios.length > 0
      ? <View items={portfolios} />
      : <Loader />;
  }
}

export default Portfolios;
