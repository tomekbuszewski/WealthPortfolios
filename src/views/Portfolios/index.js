import * as React from 'react';
import { Link } from 'react-router-dom';
import Props from 'prop-types';

import Section from '@src/components/Section';

const Portfolios = ({ items }) => (
  <Section title="Portfolios">
    <ul>
      {items.map(item => (
        <li key={item.key}><Link to={`${item.to}`}>{item.clientName}</Link></li>
      ))}
    </ul>
  </Section>
);

Portfolios.propTypes = {
  items: Props.arrayOf(Props.shape({
    to: Props.string,
    clientName: Props.string,
  })).isRequired,
};

export default Portfolios;
