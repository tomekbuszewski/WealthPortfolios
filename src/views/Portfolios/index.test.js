import * as React from 'react';
import renderer from 'react-test-renderer';
import { HashRouter } from 'react-router-dom';

import Portfolios from './';

import data from '@mocks/portfolios';

describe ('Portfolios tests', () => {
  it ('should match snapshot', () => {
    const component = renderer.create(<HashRouter>
      <Portfolios
        items={data}
      />
    </HashRouter>);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
