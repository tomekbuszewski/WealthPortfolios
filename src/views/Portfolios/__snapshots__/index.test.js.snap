// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolios tests should match snapshot 1`] = `
<section
  className="sc-bdVaJa gPOcxX section hero"
>
  <header
    className="title"
  >
    Portfolios
  </header>
  <div
    className="content is-large"
  >
    <ul>
      <li>
        <a
          href="#/undefined"
          onClick={[Function]}
        >
          Kowalski
        </a>
      </li>
      <li>
        <a
          href="#/undefined"
          onClick={[Function]}
        >
          Kowalski
        </a>
      </li>
      <li>
        <a
          href="#/undefined"
          onClick={[Function]}
        >
          Nowak
        </a>
      </li>
      <li>
        <a
          href="#/undefined"
          onClick={[Function]}
        >
          Nowak
        </a>
      </li>
      <li>
        <a
          href="#/undefined"
          onClick={[Function]}
        >
          Piotrowski
        </a>
      </li>
    </ul>
  </div>
</section>
`;
